lib/typing/ext/time.go

```diff
@@ -51,29 +52,23 @@ type ExtendedTime struct {
 	nestedKind NestedKind
 }
 
+// MarshalJSON is a custom JSON marshaller for ExtendedTime.
 func (e ExtendedTime) MarshalJSON() ([]byte, error) {
-	return json.Marshal(e.String(""))
+	return json.Marshal(e.ts)
```

Comment:
Is this not taking the NestedKind's type and format into account?

Reply:
Ah yeah. I'm purposefully moving away from using the ExtendedTime's nested type and format into account.

My goal is to eventually make ExtendedTime just an alias for `time.Time`.

MarshalJSON was created specifically for nested MongoDB objects and this code is consistent with how Mongo marshals Datetime https://github.com/mongodb/mongo-go-driver/blob/c91b204cc72025ea0c8d0845b84816b8c9e88903/bson/primitive.go#L59-L61

Reply:
Actually, let me get it to look identical to MongoDB's